src/dlgPackageExporter.cpp

```diff
@@ -136,15 +150,40 @@ void dlgPackageExporter::recurseTree(QTreeWidgetItem* pItem, QList<QTreeWidgetIt
     }
 }
 
+bool dlgPackageExporter::writeFileToZip(const QString& archiveFileName, const QString& fileSystemFileName, zip* archive)
+{
+    struct zip_source* s = zip_source_file(archive, fileSystemFileName.toUtf8().constData(), 0, -1);
+    if (s == nullptr) {
+        displayResultMessage(tr("Failed to source file \"%1\" to place into module (archive) file, error is: \"%3\".")
+                             .arg(fileSystemFileName, QString::fromUtf8(zip_strerror(archive))), false);
+        return false;
+    }
+
+    if (zip_file_add(archive, archiveFileName.toUtf8().constData(), s, ZIP_FL_ENC_UTF_8|ZIP_FL_OVERWRITE) == -1) {
+        displayResultMessage(tr("Failed to add file \"%1\" to module (archive) file \"%2\", error message was: \"%3\".")
+                             .arg(archiveFileName, mPackagePathFileName, QString::fromUtf8(zip_strerror(archive))), false);
+        return false;
+    }
+
+    return true;
+}
 
 void dlgPackageExporter::slot_export_package()
 {
-    QFile checkWriteability(filePath);
+    // Prevent a second call by removing this button:
+    ui->buttonBox->removeButton(mExportButton);
+    // Also disable the addFiles button:
+    ui->addFiles->setEnabled(false);
+
+    QFile checkWriteability(mXmlPathFileName);
     if (!checkWriteability.open(QIODevice::WriteOnly)) {
-        ui->infoLabel->setText(tr("Failed to export - couldn't open %1 for writing in. Do you have the necessary permissions to write to that folder?").arg(filePath));
+        displayResultMessage(tr("Failed to export - could not open %1 for writing in.</p>"
```

Comment:
Closing tag needs opening tag, right?

Reply:
Nope - not in this case! The outermost tags are already implicit/included in the `displayResultMessage(`...`)` method.

Reply:
Fair enough, that was unexpected. :+1: